lib/readline.js

```diff
@@ -500,6 +501,28 @@ Interface.prototype._insertString = function(c) {
     // a hack to get the line refreshed if it's needed
     this._moveCursor(0);
   }
+  // emit current line for generating preview
+  this.emit('buffer', this.line);
+};
+
+Interface.prototype.appendPreview = function(result) {
```

Comment:
Are these function meant to be added to the prototype? If so, they should be documented.

Reply:
I have moved them to `_ appendPreview`, for which I guess docs are not needed. Yes, still it can be accessible in the prototype, but since this method is needed in other place (repl), I don't know how should I expose it (given the fact `_tabComplete` etc are working in this model, so should be fine).